crd-generator/api/src/main/java/io/fabric8/crd/generator/decorator/ResourceProvidingDecorator.java

```diff
@@ -15,6 +15,25 @@
  */
 package io.fabric8.crd.generator.decorator;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public abstract class ResourceProvidingDecorator<T> extends Decorator<T> {
 
+  protected Map<String, String> toMap(String[] arr) {
+    Map<String, String> res = new HashMap<>();
+    if (arr != null) {
+      for (String e : arr) {
+        String[] splitted = e.split("\\=");
+        if (splitted.length == 2) {
+          res.put(splitted[0], splitted[1]);
+        } else {
+          throw new IllegalArgumentException(
+              "Invalid value: " + e + " cannot be parsed as a key-value pair. Expected format is 'key=value'.");
+        }
+      }
+    }
+    return res;
+  }
```

Comment:
I'm not sure about the implications of this method and the opposite `CustomResourceInfo.toStringArray`

What would happen if there's a label or annotation similar to:

```java
String [] test = new String[] {
  "edge.example.com/math= 1 + 1 = 2",
  "edge.example.com/annoying= =======> () <======"
};

// Or for the opposite method
Map.of(
  "edge.example.com/math", "1 + 1 = 2",
  "edge.example.com/annoying", "=======> () <======"
};

```

What are the implications? is this a possible scenario?